xarray/testing.py

```diff
@@ -82,7 +90,10 @@ def assert_identical(a, b):
         assert a.name == b.name
         assert_identical(a._to_temp_dataset(), b._to_temp_dataset())
     elif isinstance(a, (xr.Dataset, xr.Variable)):
-        assert a.identical(b), '{}\n{}'.format(a, b)
+        assert a.identical(b), (
+            '{}\n{}\n---\n{}\n{}'
+            .format(a, b, _get_info_as_str(a), _get_info_as_str(b))
+        )
```

Comment:
if this is a `xr.Variable` instance, this will not work since it won't have the info method.